client-java/src/main/java/uk/gov/gchq/palisade/client/JavaClient.java

```diff
@@ -0,0 +1,139 @@
+ * test class annotated with {@code MicronautTest} as it is configured for
+ * dependency injection. This makes testing much easier.
+ *
+ * @since 0.5.0
+ */
+public class JavaClient implements Client {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JavaClient.class);
+
+    private final Configuration configuration;
+    private final DownloadManager downloadManager;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Create a new JavaClient injected with required services.
+     *
+     * @param configuration   The global configuration
+     * @param downloadManager The download manager maintaining all downloads and
+     *                        thread pools
+     * @param objectMapper    The object mapper
+     */
+    public JavaClient(
+            final Configuration configuration,
+            final DownloadManager downloadManager,
+            final ObjectMapper objectMapper) {
+        this.configuration = checkArgument(configuration);
+        this.objectMapper = checkArgument(objectMapper);
+        this.downloadManager = checkArgument(downloadManager);
+    }
+
+    Configuration getConfiguration() {
+        return this.configuration;
+    }
+
+    @Override
+    public Result resume(final Path path) {
+        return resume(path, Map.of());
+    }
+
+    @Override
+    public Result resume(final Path path, final Map<String, Object> configuration) {
+        var state = new JobStateService(objectMapper).createFrom(path, configuration);
+        var receiver = createReceiver(state.getJobConfig().getReceiverClass());
+        return submit(state, receiver);
+    }
+
+    @Override
+    @SuppressWarnings("java:S3242") // I REALLY want to use UnaryOperator here SonarQube!!!
+    public Result submit(final UnaryOperator<JobRequest.Builder> func) {
+        return submit(func.apply(JobRequest.builder()).build());
+    }
+
+    @Override
+    public Result submit(final IJobRequest jobRequest) {
+
+        LOGGER.debug("Job configuration submitted: {}", jobRequest);
+
+        var receiver = createReceiver(jobRequest.getReceiverClass());
+        var stateService = new JobStateService(objectMapper);
+        var state = stateService.createNew(jobRequest, configuration);
+
+        return submit(state, receiver);
+    }
+
+    private Result submit(final JobState state, final Receiver receiver) {
+
+        var palisadeService = new PalisadeService(objectMapper, state.getConfiguration().getPalisadeUri());
+        var eventBus = new EventBus();
+
+        var job = ClientJob.createJob(b -> b
+            .state(state)
+            .palisadeClient(palisadeService)
+            .receiver(receiver)
+            .eventBus(eventBus)
+            .downloadManager(downloadManager)
+            .objectMapper(objectMapper)
+        );
+        return job.start();
+    }
+
+    private static final Receiver createReceiver(final Class<?> receiverClass) {
```

Comment:
Nothing to do with inheritance. `final` just means that the parameter's reference cannot be reassigned within the method. I wish final was the default in Java. Good to have final as otherwise a parameter cannot be used inside a lambda.